### PR TITLE
112 check tags

### PR DESCRIPTION
**Pull request by milldr**
Debugging  #112 in progress.

This PR includes:
- Testing for builtin tf12 rules
- Added field to BuiltInTestCase struct for giving tf version. Propagated to all existing tests cases. Valid tests that already work for tf12 given "both" tag so that they will test both terraform versions
- Added test for checking for tags as a custom TF12 rules, as the bug describes in the issue

### CODE REVIEW

**Comment by milldr** [comment]
Also includes more verbose error messages for test failures in both terraform builtin tests and in tf12 custom rule tests

File: linter/terraform_v12_test.go [hunk 1]
```diff
@@ -195,6 +196,27 @@ func TestTerraform12ModuleFileName(t *testing.T) {
...
+  var violationsReported string
+  for count, v := range violations {
+    violationsReported += strconv.Itoa(count + 1) + ". Violation:"
+    violationsReported += "\n\tRule Message: " + v.RuleMessage
+    violationsReported += "\n\tRule Id: " + v.RuleID
+    violationsReported += "\n\tResource ID: " + v.ResourceID
+    violationsReported += "\n\tResource Type: " + v.ResourceType
+    violationsReported += "\n\tCategory: " + v.Category
+    violationsReported += "\n\tStatus: " + v.Status
+    violationsReported += "\n\tRule Message: " + v.RuleMessage
```
Comment (phelewski) [inline linter/terraform_v12_test.go]: You have this `RuleMessage` line printed twice in here.

File: linter/terraform_v12_test.go [hunk 2]
```diff
@@ -353,6 +375,12 @@ func TestTerraform12LinterCases(t *testing.T) {
 			1,
 			"NO_SSH_ACCESS",
 		},
+		"TF12Tagging": {
+      "./testdata/resources/tagging.tf",
```
Comment (phelewski) [inline linter/terraform_v12_test.go]: spacing

File: linter/terraform_v12_test.go [hunk 3]
```diff
@@ -371,7 +399,8 @@ func TestTerraform12LinterCases(t *testing.T) {
 		}
 		if len(report.Violations) != tc.ExpectedViolationCount {
 			t.Errorf("%s returned %d violations, expecting %d", name, len(report.Violations), tc.ExpectedViolationCount)
-			t.Errorf("Violations: %v", report.Violations)
+      violationsReported := getViolationsString(report.Violations)
```
Comment (phelewski) [inline linter/terraform_v12_test.go]: spacing

File: linter/testdata/rules/tagging.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+            value: public,internal,confidential
+          # Fuzzy matching for nested modules that cant be resolved
+          - key: tags
+            op: contains
+            value: tags
+          - key: tags
+            op: contains
+            value: TAGS
+
+            #     assertions:
```
Comment (phelewski) [inline linter/testdata/rules/tagging.yml]: This can be cleaned up

File: cli/builtin_terraform_test.go [hunk 5]
```diff
@@ -3,6 +3,7 @@ package main
 import (
 	"fmt"
 	"testing"
+  "strconv"
```
Comment (phelewski) [inline cli/builtin_terraform_test.go]: spacing

File: cli/builtin_terraform_test.go [hunk 6]
```diff
@@ -16,10 +17,11 @@ func loadRules(t *testing.T, filename string) assertion.RuleSet {
 }
 
 type BuiltInTestCase struct {
-	Filename     string
-	RuleID       string
-	WarningCount int
-	FailureCount int
+  TerraformVersion  string // "tf", "tf12", or "both"
```
Comment (phelewski) [inline cli/builtin_terraform_test.go]: spacing

File: cli/builtin_terraform_test.go [hunk 7]
```diff
@@ -41,103 +43,152 @@ func numberOfFailures(violations []assertion.Violation) int {
...
+  for count, v := range violations {
+    if v.Status == severity {
+      violationsReported += strconv.Itoa(count + 1) + ". Violation:"
+      violationsReported += "\n\tRule Message: " + v.RuleMessage
+      violationsReported += "\n\tRule Id: " + v.RuleID
+      violationsReported += "\n\tResource ID: " + v.ResourceID
+      violationsReported += "\n\tResource Type: " + v.ResourceType
+      violationsReported += "\n\tCategory: " + v.Category
+      violationsReported += "\n\tStatus: " + v.Status
+      violationsReported += "\n\tRule Message: " + v.RuleMessage
```
Comment (phelewski) [inline cli/builtin_terraform_test.go]: `RuleMessage` printed out twice here

**Comment by milldr** [comment]
1. removed duplicate rule messages
2. cleaned test case commented code
3. retabbed both modified go files

**Pull request merged**